docs/api-conventions.md

```diff
@@ -3,32 +3,57 @@ API Conventions
+* **Resource** a representation of a system entity, sent or retrieved as JSON via HTTP to the server. Resources are exposed via:
+  * Collections - a list of resources of the same type, which may be queryable
+  * Elements - an individual resource, addressible via a URL
 
-Types of Resources
-------------------
+Each resource typically accepts and returns data of a single kind.  A kind may be accepted or returned by multiple resources that reflect specific use cases. For instance, the kind "pod" is exposed as a "pods" resource that allows end users to create, update, and delete pods, while a separate "pod status" resource (that acts on "pod" kind) allows automated processes to update a subset of the fields in that resource. A "restart" resource might be exposed for a number of different resources to allow the same action to have different results for each object.
+
+
+Types (Kinds)
+-------------
+
+Kinds are grouped into three categories:
+
+1. **Objects** represent a persistent entity in the system.
+
+   Creating an API object is a record of intent - once created, the system will work to ensure that resource exists. All API objects have common metadata.
+
+   An object may have multiple resources that clients can use to perform specific actions than create, update, delete, or get.
 
-All API resources are either:
+   Examples: Pods, ReplicationControllers, Services, Namespaces, Nodes
 
-1. **Objects** represents a physical or virtual construct in the system. An API object is a record of intent - once created, the system will work to ensure that resource exists. All API objects have common metadata intended for client use.
-2. **Lists** are collections of **objects** of one or more types. Lists have a limited set of common metadata. All lists use the "items" field to contain the array of objects they return. Each resource kind should have an endpoint that returns the full set of resources, as well as zero or more endpoints that return subsets of the full list.
+2. **Lists** are collections of **resources** of one (usually) or more (occasionally) kinds.
+
+   Lists have a limited set of common metadata. All lists use the "items" field to contain the array of objects they return.
+
+   Most objects defined in the system should have an endpoint that returns the full set of resources, as well as zero or more endpoints that return subsets of the full list. Some objects may be singletons (the current user, the system defaults) and may not have lists.
 
    In addition, all lists that return objects with labels should support label filtering (see [labels.md](labels.md), and most lists should support filtering by fields.
 
+   Examples: PodLists, ServiceLists, NodeLists
+
    TODO: Describe field filtering below or in a separate doc.
 
+3. **Simple** kinds are used for specific actions on objects and for non-persistent entities.
+
+   Given their limited scope, they have the same set of limited common metadata as lists.
```

Comment:
ListMeta (poorly named though it is), seems appropriate.  Status doesn't really have a name or namespace or labels or annotations, so I think our initial call on ListMeta vs Object was accurate.  Maybe we rename ListMeta to SimpleMeta eventually.